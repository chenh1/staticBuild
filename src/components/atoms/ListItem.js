import React, {PropTypes}  from 'react';
import { Link, IndexLink } from 'react-router';

const ListItem = ({text, path, linkClass, linkActiveClass}) => {
    return (
        <li>
            {path ? 
                <Link to={path} className={linkClass} activeClassName={linkActiveClass} target={text === 'Resume' ? '_blank': ''}>{text}</Link>
                :
                <span>{text}</span>
            }
        </li>
    );
};

ListItem.propTypes = {
    text: PropTypes.string.isRequired,
    path: PropTypes.string,
    linkClass: PropTypes.string,
    linkActiveClass: PropTypes.string
};

export default ListItem;