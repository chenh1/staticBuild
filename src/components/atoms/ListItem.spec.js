import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import ListItem from './ListItem';

test('renders list item with link if path prop exists', () => {
    const tree = renderer.create(<ListItem
        text="Mock list item"
        path="/mockPath"
        linkClass="mock-class"
        linkActiveClass="mock-active-class" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test('renders list item with text in span tag only if path does not exist', () => {
    const tree = renderer.create(<ListItem
        text="Mock list item" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});