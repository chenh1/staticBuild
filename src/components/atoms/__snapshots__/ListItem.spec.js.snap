// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list item with link if path prop exists 1`] = `
<li>
  <a
    className="mock-class"
    onClick={[Function]}
    style={Object {}}
    target=""
  >
    Mock list item
  </a>
</li>
`;

exports[`renders list item with text in span tag only if path does not exist 1`] = `
<li>
  <span>
    Mock list item
  </span>
</li>
`;
